File: android/core/domain/src/main/kotlin/com/globussoft/wellness/core/domain/model/TreatmentPlan.kt
package com.globussoft.wellness.core.domain.model

/**
 * A course-of-treatment plan linking a patient, a service, and a session quota.
 *
 * [completedSessions] is derived server-side from linked visit records.
 * [status] reflects ACTIVE / PAUSED / CANCELLED / COMPLETED lifecycle.
 */
data class TreatmentPlan(
    val id: String,
    val name: String,
    val patientId: String,
    val serviceName: String?,
    val totalSessions: Int,
    val completedSessions: Int,
    val totalPrice: Double?,
    val status: String,
    val startedAt: String?,
    val nextDueAt: String?,
)

File: iOS/GlobusCRM/GlobusCRM/Features/Health/Domain/Models/TreatmentPlan.swift
import Foundation

struct TreatmentPlan: Identifiable, Equatable {
    let id: String
    let name: String
    let serviceName: String?
    let serviceCategory: String?
    let startedAt: String?
    let nextDueAt: String?
    let sessionsTotal: Int
    let sessionsCompleted: Int
    let status: String
    let totalPrice: Double?

    var progressFraction: Double {
        guard sessionsTotal > 0 else { return 0 }
        return Double(sessionsCompleted) / Double(sessionsTotal)
    }
}
